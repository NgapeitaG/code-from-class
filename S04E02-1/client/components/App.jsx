import React from 'react'

import Counter from './Counter'

const App = () => {
  return (
    <div>
      <h1>React development has begun!</h1>
      <Counter />
    </div>
  )
}

export default App
