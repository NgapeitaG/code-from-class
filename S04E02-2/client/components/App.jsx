import React from 'react'

import Colors from './Colors'

const App = () => {
  return (
    <div>
      <h1>Colours are cool!</h1>
      <Colors />
    </div>
  )
}

export default App
