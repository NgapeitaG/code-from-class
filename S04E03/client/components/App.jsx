import React from 'react'
import {Route} from 'react-router-dom'

import Home from './Home'
import Animal from './Animal'
import Animals from './Animals'

const animals = ['bear', 'pig', 'eagle']

const App = () => {
  return (
    <div>
      <h1>Animal App</h1>
      <Route exact path="/" component={Home} />
      <Route exact path="/animals" render={() => (
        <Animals animals={animals} />
      )} />
      <Route path="/animals/:animal" component={Animal} />
    </div>
  )
}

export default App
