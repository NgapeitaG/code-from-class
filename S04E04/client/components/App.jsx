import React from 'react'
import {Route} from 'react-router-dom'

import List from './List'

const App = () => {
  return (
    <div>
      <h1>React form demo</h1>
      <Route path="/" component={List} />
    </div>
  )
}

export default App
