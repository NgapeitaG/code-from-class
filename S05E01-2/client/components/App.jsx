import React from 'react'

import APIError from './APIError'
import Routes from './Routes'
import { getPosts } from '../api'

export default class App extends React.Component {
  state = {
    posts: [],
    message: null
  }

  componentDidMount () {
    this.fetchPosts()
  }

  fetchPosts = () => getPosts()
    .then(posts => this.setState({ posts }))
    .catch(({ message }) => this.setState({ message }))

  render () {
    const { message, posts } = this.state

    if (message) {
      return <APIError message={message} />
    }

    return (
      <React.Fragment>
        <Routes posts={posts} fetchPosts={this.fetchPosts} />
      </React.Fragment>
    )
  }
}
