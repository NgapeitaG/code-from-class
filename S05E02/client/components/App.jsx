import React from 'react'
import request from 'superagent'

class App extends React.Component {
  // Alternative class syntax
  state = {
    message: ''
  }

  componentDidMount () {
    request.post('http://localhost:3000')
      .send({
        greeting: 'Hello world!'
      })
      .then(res => {
        this.setState({
          message: res.body.message
        })
      })
      .catch(err => {
        this.setState({
          message: err.message
        })
      })
  }

  render () {
    return (
      <React.Fragment>
        <h1>React development has begun!</h1>
        <div>The server said:</div>
        <div>{this.state.message}</div>
      </React.Fragment>
    )
  }
}

export default App
