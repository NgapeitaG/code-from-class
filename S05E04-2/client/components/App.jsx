import React from 'react'

import {getPerson} from '../api'

class App extends React.Component {
  constructor () {
    super()

    this.state = {
      name: ''
    }
  }

  componentDidMount () {
    const personId = 1
    getPerson(personId)
      .then(person => {
        this.setState({name: person.name})
      })
      .catch(err => {
        console.error('Error:', err)
      })
  }

  render () {
    return (
      <div>
        <h1>In a galaxy far, far away ...</h1>
        <div>Name: {this.state.name}</div>
      </div>
    )
  }
}

export default App
