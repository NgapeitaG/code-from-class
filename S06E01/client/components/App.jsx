import React from 'react'

import Wombats from './Wombats'

function App (props) {
  const wombats = props.wombats
  return (
    <div>
      <Wombats wombats={wombats} />
    </div>
  )
}

export default App
