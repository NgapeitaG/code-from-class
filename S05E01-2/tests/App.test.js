import React from 'react'
import App from '../client/components/App'

import { shallow } from 'enzyme'

test('App matches the last snapshot', () => {
  expect(shallow(<App />)).toMatchSnapshot()
})
