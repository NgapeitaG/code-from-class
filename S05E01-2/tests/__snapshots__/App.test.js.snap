// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sidebar pure-u-1 pure-u-md-1-4"
        >
          <Header />
        </div>,
        <div
          className="content pure-u-1 pure-u-md-3-4"
        >
          <Routes
            fetchPosts={[Function]}
            posts={Array []}
          />
          
        </div>,
        <Footer />,
      ],
      "className": "pure-g",
      "id": "layout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header />,
          "className": "sidebar pure-u-1 pure-u-md-1-4",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Routes
              fetchPosts={[Function]}
              posts={Array []}
            />,
            "",
          ],
          "className": "content pure-u-1 pure-u-md-3-4",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fetchPosts": [Function],
              "posts": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sidebar pure-u-1 pure-u-md-1-4"
          >
            <Header />
          </div>,
          <div
            className="content pure-u-1 pure-u-md-3-4"
          >
            <Routes
              fetchPosts={[Function]}
              posts={Array []}
            />
            
          </div>,
          <Footer />,
        ],
        "className": "pure-g",
        "id": "layout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header />,
            "className": "sidebar pure-u-1 pure-u-md-1-4",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Routes
                fetchPosts={[Function]}
                posts={Array []}
              />,
              "",
            ],
            "className": "content pure-u-1 pure-u-md-3-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fetchPosts": [Function],
                "posts": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
